### PR TITLE
Added story_separator

### PR DESCRIPTION
**Pull request by Fliens**
Added story_separator
By defining the separator the text will be split by the separator rather than using the story_start_words or story_words_per_epoch
Any str can be used as a separator! If the separator is not in the text, the separator will be ignored and create_story will work as usual
Also updated the README.md to include the change

### CODE REVIEW

**Comment by NotNANtoN** [comment]
This looks good implementation-wise. I'm just wondering whether it would be possible better to exclude the separator from the generation. Your example works well for normal punctuation, but if someone would want some weird splitting, then a special character such as `|` should probably be used - which you do not want deepdaze to interpret, therefore the separator would need to be removed from self.words.

Of course, this makes things more complicated. Maybe it would not be a problem to pass sentences into deepdaze as e.g. `I am a chicken man. I have wings. |`. But I have not tried it and would be careful.

**Comment by Fliens** [comment]
> This looks good implementation-wise. I'm just wondering whether it would be possible better to exclude the separator from the generation. Your example works well for normal punctuation, but if someone would want some weird splitting, then a special character such as `|` should probably be used - which you do not want deepdaze to interpret, therefore the separator would need to be removed from self.words.
> 
> Of course, this makes things more complicated. Maybe it would not be a problem to pass sentences into deepdaze as e.g. `I am a chicken man. I have wings. |`. But I have not tried it and would be careful.

@NotNANtoN 
The separator will now be removed in each epoch but also in the filename

**Comment by Fliens** [comment]
My question is if there should be some kind of testing if the user gives an empty string or one with the separator only
I didn't implement that since it's not in the normal mode (without --create_story)

**Comment by NotNANtoN** [comment]
I assume some testing does not hurt and makes for a better user experience

File: deep_daze/deep_daze.py [hunk 1]
```diff
@@ -308,7 +314,13 @@ def __init__(
...
             # add one epoch if not divisible
             self.epochs = int(self.epochs) if int(self.epochs) == self.epochs else int(self.epochs) + 1
-            print("Running for ", self.epochs, "epochs")
+            if self.separator is not None:
+                if self.separator not in text:
+                    print("Separator '"+self.separator+"' will be ignored since not in text!")
+                    self.separator = None
+                else:
+                    self.epochs = len(list(filter(None,text.split(self.separator))))
+            print("Running for", self.epochs, "epochs" + (" (splitted with '"+self.separator+"' as the separator)" if self.separator is not None else ""))
```
Comment (StevenUS) [inline deep_daze/deep_daze.py:323]: recommend changing "splitted" to "split"

Comment (Fliens) [reply deep_daze/deep_daze.py:323]: Oh, you're right

**Comment by Fliens** [comment]
@StevenUS Could you please look over the sentences that I've now added since I'm not 100% sure if they're completely right :)

**Comment by StevenUS** [comment]
@Fliens looks good. One suggestion would be to move the trailing comments to the line above, to match the rest of the file/project.

**Comment by lucidrains** [comment]
lgtm 🙏

**Pull request merged**